### PR TITLE
[dag] sign metadata and rename signature to vote

### PR DESCRIPTION
**Pull request by zekun000**
Now we're signing the metadata instead of only the digest itself, so that we can verify the parents links correctly. Also rename the signature to Vote hopefully more understandable.

### CODE REVIEW

File: consensus/src/dag/reliable_broadcast.rs [hunk 1]
```diff
@@ -183,53 +179,51 @@ impl NodeBroadcastHandler {
...
-                let signature = node.sign(&self.signer)?;
-                let digest_signature =
-                    NodeDigestSignature::new(node.metadata().epoch(), node.digest(), signature);
+                let signature = node.sign_vote(&self.signer)?;
+                let vote = Vote::new(node.metadata().clone(), signature);
 
-                self.storage
-                    .save_node_signature(&NodeId::from(&node), &digest_signature)?;
-                signatures_by_peer.insert(*node.metadata().author(), digest_signature.clone());
+                self.storage.save_vote(&node.id(), &vote)?;
```
Comment (sasha8) [inline consensus/src/dag/reliable_broadcast.rs:223]: @ibalajiarun, one less from :)

**Review by sasha8 — APPROVED** [review]

File: consensus/src/dag/reliable_broadcast.rs [hunk 2]
```diff
@@ -183,53 +179,51 @@ impl NodeBroadcastHandler {
...
                 .entry(node_id.round())
                 .or_insert_with(BTreeMap::new)
-                .insert(node_id.author(), node_sig);
+                .insert(node_id.author(), vote);
         } else {
             to_delete.push(node_id);
         }
     }
-    if let Err(err) = storage.delete_node_signatures(to_delete) {
+    if let Err(err) = storage.delete_vote(to_delete) {
```
Comment (ibalajiarun) [inline consensus/src/dag/reliable_broadcast.rs:200]:
```suggestion
    if let Err(err) = storage.delete_votes(to_delete) {
```
delete_votes plural?

File: consensus/src/dag/types.rs [hunk 3]
```diff
@@ -241,9 +222,16 @@ impl Node {
...
     }
 
-    pub fn sign(&self, signer: &ValidatorSigner) -> Result<Signature, CryptoMaterialError> {
-        let node_digest = NodeDigest::new(self.digest());
-        signer.sign(&node_digest)
+    pub fn epoch(&self) -> u64 {
+        self.metadata.epoch
+    }
+
+    pub fn id(&self) -> NodeId {
```
Comment (ibalajiarun) [inline consensus/src/dag/types.rs:229]: We can remove the From<&Node> for NodeId impl.

File: consensus/src/dag/reliable_broadcast.rs [hunk 4]
```diff
@@ -166,13 +166,9 @@ impl NodeBroadcastHandler {
...
-                missing_parents.iter().all(|parent| {
-                    let node_digest = NodeDigest::new(*parent.metadata().digest());
-                    self.epoch_state
-                        .verifier
-                        .verify_multi_signatures(&node_digest, parent.signatures())
-                        .is_ok()
-                }),
+                missing_parents
+                    .iter()
+                    .all(|parent| { parent.verify(&self.epoch_state.verifier).is_ok() }),
```
[the remark this answers is not in the document]
Comment (ibalajiarun) [reply consensus/src/dag/reliable_broadcast.rs:171]: Node.verify( ) checks for this. We separated validations from self-contained checks.

**Review by ibalajiarun — APPROVED** [review]

**Pull request merged**